From f78424644f405044569e8b95b259750afa64d605 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 11 Jan 2024 17:01:42 +0100
Subject: [PATCH 0421/1695] drivers: i2c: add missing __unused in stubbed
 function

Fix statc inline stub implementation of i2c_dt_get_dev() that
lacks a __unused attribute on an unused argument.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2cafb6a4cde26bee7bd257fd4cc069fb2b7bf9a9 drivers: i2c: add missing __unused in stubbed function]
Change-Id: If6c11b036f7236cedd1018359e4fe114d209bd90
Fixes: b357d34fe91f ("core: dt_driver: swap TEE_result and retrieved device reference")
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/include/drivers/i2c.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/include/drivers/i2c.h b/core/include/drivers/i2c.h
index 1cc9cdaef..287689cc1 100644
--- a/core/include/drivers/i2c.h
+++ b/core/include/drivers/i2c.h
@@ -221,7 +221,7 @@ static inline TEE_Result i2c_smbus_raw(struct i2c_dev *i2c_dev __unused,
 
 static inline TEE_Result i2c_dt_get_dev(const void *fdt __unused,
 					int nodeoffset __unused,
-					struct i2c_dev **i2c_dev)
+					struct i2c_dev **i2c_dev __unused)
 {
 	return TEE_ERROR_NOT_SUPPORTED;
 }
-- 
2.47.2

